.changeset/thirty-beds-smoke.md

```diff
@@ -0,0 +1,8 @@
+---
+"@astrojs/mdx": minor
+"@astrojs/markdown-remark": minor
+---
+
+Handle syntax highlighting using rehype plugins instead of remark plugins. This provides better interoperability with other [rehype plugins](https://github.com/rehypejs/rehype/blob/main/doc/plugins.md#list-of-plugins) that deal with code blocks, in particular with third party syntax highlighting plugins and [`rehype-mermaid`](https://github.com/remcohaszing/rehype-mermaid).
```

Comment:
```suggestion
Changes Astro's internal syntax highlighting to use rehype plugins instead of remark plugins. This provides better interoperability with other [rehype plugins](https://github.com/rehypejs/rehype/blob/main/doc/plugins.md#list-of-plugins) that deal with code blocks, in particular with third party syntax highlighting plugins and [`rehype-mermaid`](https://github.com/remcohaszing/rehype-mermaid).
```

Given that this is a *change to an existing thing*, I like phrasing this so it makes it clear upfront that something has changed! (and especially given that below we're going to say your project might actually break)

Make sure the nuance is correct here, and correct as necessary, but I like the "in your face-ness" of "We changed how Astro's syntax highlighting works!"